rpcserver/router.go

```diff
@@ -778,6 +778,16 @@ func (server *routedBoltzServer) createSwap(ctx context.Context, isAuto bool, re
 
 	if swapResponse == nil {
 
+		if createSwap.Invoice != "" {
+			existing, err := server.database.QuerySwapByInvoice(createSwap.Invoice)
```

Comment:
It would be better to query by preimage hash

Reply:
we dont have a preimage hash column in the db - think its fine this way.